.github/workflows/Build.yml

```diff
@@ -48,12 +48,17 @@ jobs:
     name: Generate Code Coverage
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
-      - uses: actions-rust-lang/setup-rust-toolchain@v1
       - name: read threshold file into environment variable
         run: |
+          git init
+          git clone https://github.com/rdkcentral/Ripple.git
```

Comment:
In git there are options to juust get that branch alone as a single clone instead of the full set

git clone <url> --branch <branch> --single-branch <folder>

we can avoid 4 steps with 1